llvm/lib/IR/Verifier.cpp

```diff
@@ -5235,6 +5235,23 @@ void Verifier::visitIntrinsicCall(Intrinsic::ID ID, CallBase &Call) {
     }
     break;
   }
+  case Intrinsic::ucmp:
+  case Intrinsic::scmp: {
+
+    Type *SrcTy = Call.getOperand(0)->getType();
+    Type *DestTy = Call.getType();
+
+    Check(DestTy->getScalarSizeInBits() >= 2, "DestTy must be at least 2 bits wide", Call);
+
+    bool isDestTypeVector = DestTy->isVectorTy();
+    Check(SrcTy->isVectorTy() == isDestTypeVector,
+        "[u]scmp source and destination must both be a vector or neither", Call);
+    if (isDestTypeVector) {
+      Check(SrcTy->getArrayNumElements() == DestTy->getArrayNumElements(),
```

Comment:
Array != vector. The way to get the number of vector elements is `cast<VectorType>(SrcTy)->getElementCount() `. It looks like there is no convenience method directly on Type...

Reply:
Hmph, fixing that here, but maybe I can write that down as a separate TODO for later.